Makefile

```diff
@@ -114,6 +114,79 @@ export ROCM_LDFLAGS := -L$(ROCM_LIBDIR) -lamdhip64
 export ROCM_TEST_CXXFLAGS := -DGPU_DEBUG
 endif
 
+# Intel oneAPI
+
+# Compiler flags supported by GCC but not by the LLVM-based compilers (clang, hipcc, dpcpp, etc.)
+LLVM_UNSUPPORTED_CXXFLAGS := --param vect-max-version-for-alias-checks=50 -Werror=format-contains-nul -Wno-non-template-friend -Werror=return-local-addr -Werror=unused-but-set-variable
+
+# flags to compile AOT:
+AOT_INTEL_FLAGS   := -fsycl-targets=spir64_x86_64,spir64_gen -Xsycl-target-backend=spir64_gen "-device 0x020a"
+AOT_CUDA_FLAGS    := -fsycl-targets=nvptx64-nvidia-cuda -fno-bundle-offload-arch --cuda-path=$(CUDA_BASE) -Wno-unknown-cuda-version -Wno-linker-warnings
+AOT_HIP_FLAGS     := -fsycl-targets=amdgcn-amd-amdhsa -Xsycl-target-backend --offload-arch=gfx900 --rocm-path=$(ROCM_BASE) -Wno-linker-warnings 
+
+# INTEL flags: compile AOT for all the CPUs and for the GPU on olice-05
+# CUDA flags : compile AOT for for NVIDIA GPUs
+# HIP flags  : compile AOT for architectures with ID gfx900 (e.g. the Radeon PRO WX 9100)
+
+# -Wno-linker-warnings will not be needed be needed anymore with https://github.com/intel/llvm/pull/7245
+
+ifdef USE_SYCL_ONEAPI
+ONEAPI_BASE       := /cvmfs/projects.cern.ch/intelsw/oneAPI/linux/x86_64/2022
+ONEAPI_VERSION    := latest
+TBB_BASE          := $(ONEAPI_BASE)/tbb/latest
+TBB_LIBDIR        := $(TBB_BASE)/lib/intel64/gcc4.8
+ifneq ($(wildcard $(ONEAPI_BASE)),)
+ONEAPI_ENV        := $(ONEAPI_BASE)/setvars.sh # --config="/eos/user/a/aperego/dev/pixeltrack-standalone/config.txt"
+                                               # the config.txt file can be used to source only specific tools 
+                                               # or a specific version of a tool of the oneAPI package
+SYCL_BASE         := $(ONEAPI_BASE)/compiler/$(ONEAPI_VERSION)/linux
+USER_SYCLFLAGS    := -fp-model=precise -fimf-arch-consistency=true -no-fma
```

Comment:
Is there are reason to use `USER_SYCLFLAGS` specifically? The `USER_*` variables are intended to be overridden from command line, so these would be lost. I'd suggest to use a different name for the variable (or move directly as part of `SYCL_CXXFLAGS`)

Reply:
Not really, I can move them